robottelo/products.py

```diff
@@ -933,6 +934,9 @@ def setup_virtual_machine(
                 repo['label'] for repo in self.custom_repos_info if repo['content-type'] == 'yum'
             ]
 
+        if location:
+            vm.set_facts({'locations.facts': {'foreman_location': str(location.name)}})
```

Comment:
Should I move it into `contenthost_setup` method.

Reply:
Yeah! 100% agree with a passing location in the contenthost_setup instead of setting the facts to VM from this method.

Since contenthost_setup is accepting the context for the organization as a parameter, it makes more sense to add the location context from this method.

Reply:
Implemented. Can you please re-review @jyejare?